apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java

```diff
@@ -639,57 +631,25 @@ public void setContent(eu.xenit.apix.data.NodeRef node, eu.xenit.apix.data.Conte
     @Override
     public eu.xenit.apix.data.ContentData createContentWithMimetypeGuess(InputStream inputStream, String fileName,
             String encoding) {
-        // Making copy of original inputstream because we want an inputstream that definitely supports mark/reset
-        InputStream inputStreamCopy = null;
         try {
-            inputStreamCopy = cloneInputStreamWithMarkSupported(inputStream);
-            String mimeType = guessMimetype(fileName, inputStreamCopy);
-
             ContentWriter writer = contentService.getWriter(null, ContentModel.PROP_CONTENT, false);
-            writer.setMimetype(mimeType);
+            if (writer instanceof MimetypeServiceAware) {
+                ((MimetypeServiceAware) writer).setMimetypeService(getServiceRegistry().getMimetypeService());
+            }
+            writer.guessMimetype(fileName);
```

Comment:
This also works on file content, right? Can you add a comment about that? I think people reading this will assume this only looks at the file extension otherwise.